ql/src/test/queries/clientpositive/compaction_query_based_masking.q

```diff
@@ -0,0 +1,31 @@
+--! qt:replace:/createTime:(\d+)/#Masked#/
+--! qt:replace:/location:(\S+)/#Masked#/
+--! qt:replace:/lastAccessTime:(\d+)/#Masked#/
+--! qt:replace:/ownerType:(\S*)/#Masked#/
+--! qt:replace:/owner:(\S*)/#Masked#/
+--! qt:replace:/skewedColValueLocationMaps:(\S*)/#Masked#/
+--! qt:replace:/transient_lastDdlTime=(\d+)/#Masked#/
+--! qt:replace:/totalSize=(\d+)/#Masked#/
+--! qt:replace:/rawDataSize=(\d+)/#Masked#/
+--! qt:replace:/writeId:(\d+)/#Masked#/
+--! qt:replace:/bucketing_version=(\d+)/#Masked#/
+--! qt:replace:/id:(\d+)/#Masked#/
+
+
+set hive.mapred.mode=nonstrict;
+set hive.security.authorization.enabled=true;
+set hive.security.authorization.manager=org.apache.hadoop.hive.ql.security.authorization.plugin.sqlstd.SQLStdHiveAuthorizerFactoryForTest;
+
+drop table masking_test_n_compact;
+
+create table masking_test_n_compact (key int, value string) stored as orc TBLPROPERTIES('transactional'='true');
+
+
+
+insert into masking_test_n_compact values('1', 'text1');
+insert into masking_test_n_compact values('2', 'text2');
+insert into masking_test_n_compact values('3', 'text3');
+
+alter table masking_test_n_compact compact 'MAJOR' and wait;
```

Comment:
wow, do we support compaction in qtests? cool

anyway, how does this qtest validate that the original issue is gone? it just starts a compaction and selects the data, I guess we might want to see that the result files are not masked

Reply:
Is it possible to add a example where it returns masked result as well?

Reply:
Fixed both the qtest and the production code. The test now checks properly if the data was written correctly or not: The test masking and filtering rules are applied based on the table name, so renaming the table will remove it from their scope. This way I can check before and after compaction if the table data was corrupted or not.